pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -206,25 +211,37 @@ protected void setupInitialSegments(String realtimeTableName, ZNRecord partition
     List<ZNRecord> records = new ArrayList<>(nPartitions);
     final long now = System.currentTimeMillis();
     final int seqNum = 0; // Initial seq number for the segments
+
     for (int i = 0; i < nPartitions; i++) {
-      final List instances = partitionMap.get(Integer.toString(i));
-      LLCRealtimeSegmentZKMetadata metadata = new LLCRealtimeSegmentZKMetadata();
-      final String rawTableName = TableNameBuilder.extractRawTableName(realtimeTableName);
-      LLCSegmentName llcSegmentName = new LLCSegmentName(rawTableName, i, seqNum, now);
-      final String segName = llcSegmentName.getSegmentName();
-
-      metadata.setCreationTime(now);
-      metadata.setStartOffset(startOffset);
-      metadata.setNumReplicas(instances.size());
-      metadata.setTableName(rawTableName);
-      metadata.setSegmentName(segName);
-      metadata.setStatus(CommonConstants.Segment.Realtime.Status.IN_PROGRESS);
-
-      ZNRecord record = metadata.toZNRecord();
-      final String znodePath = PinotRealtimeSegmentManager.getSegmentsPath() + "/" + realtimeTableName + "/" + segName;
-      paths.add(znodePath);
-      records.add(record);
-      idealStateEntries.put(segName, instances);
+      SimpleConsumerWrapper kafkaConsumer = SimpleConsumerWrapper.forPartitionConsumption(new KafkaSimpleConsumerFactoryImpl(),
```

Comment:
I'd rather not, otherwise that means `SimpleConsumerWrapper` will now depend on `KafkaStreamMetadata` and will now need to know about the fact that there is a stream metadata, while the current implementation is not coupled to any Pinot-specific concepts. In theory, one could take `SimpleConsumerWrapper` and drop it in another system and it would be good to keep it this way.

If we were to pass the metadata around, then we shouldn't have `SimpleConsumerWrapper` depend on it, but this also means that we should update the signature to remove some of the arguments (`realtimeTableName`, `nReplicas`) and the relevant unit tests.